### PR TITLE
Updated maxSupportedAPIVersion to 12

### PR DESCRIPTION
**Pull request by DanielLavie**
### What does this PR do?
This PR updates `maxSupportedAPIVersion` to be the higher of `KAFKA_MAX_SUPPORTED_PRODUCE_REQUEST_API_VERSION` and `KAFKA_MAX_SUPPORTED_FETCH_REQUEST_API_VERSION`.

### Motivation
Ensure that when one of the maximum API versions is updated, the maxSupportedAPIVersion value is automatically updated, preventing any oversight.

### CODE REVIEW

File: pkg/network/protocols/kafka/api_version_counter.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 
 const (
 	minSupportedAPIVersion = 1
-	maxSupportedAPIVersion = 11
+	maxSupportedAPIVersion = 12
```
Comment (guyarb) [inline pkg/network/protocols/kafka/api_version_counter.go:18]: The PR does not prevent the next time it happens
I do recommend to populate the maximum supported versions from the kernel (via `types.go` file) and select the max

Comment (DanielLavie) [reply pkg/network/protocols/kafka/api_version_counter.go]: Done

**Review by guyarb — APPROVED** [review]

File: pkg/network/protocols/kafka/api_version_counter.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 
 const (
 	minSupportedAPIVersion = 1
-	maxSupportedAPIVersion = 12
+	maxSupportedAPIVersion = max(MaxSupportedProduceRequestApiVersion, MaxSupportedFetchRequestApiVersion)
```
[the remark this answers is not in the document]
Comment (DanielLavie) [reply pkg/network/protocols/kafka/api_version_counter.go:18]: It works in go playground:
https://go.dev/play/p/iP22YYrLz1v?v=goprev

**Pull request merged**